x-pack/test/functional/services/observability/alerts/rules_page.ts

```diff
@@ -30,12 +31,127 @@ export function ObservabilityAlertsRulesProvider({ getService }: FtrProviderCont
     await find.clickByButtonText('metric-threshold');
   };
 
+  const clickOnObservabilityCategory = async () => {
+    const categories = await testSubjects.find('ruleTypeModal');
+    const category = await categories.findByCssSelector(`.euiFacetButton[title="Observability"]`);
+    await category.click();
+  };
+
+  const clickOnCustomThresholdRule = async () => {
+    await testSubjects.click('observability.rules.custom_threshold-SelectOption');
+  };
+
+  const fillCustomThresholdRule = async (name: string) => {
+    await testSubjects.setValue('ruleNameInput', name);
+    await testSubjects.setValue('comboBoxSearchInput', 'tag1');
+
+    // select data view
+    await testSubjects.click('selectDataViewExpression');
+    await testSubjects.setValue('indexPattern-switcher--input', 'metricbeat-*');
+    const dataViewExpression = await find.byCssSelector(
+      '[data-test-subj="indexPattern-switcher--input"]'
+    );
+    await dataViewExpression.pressKeys(Key.ENTER);
+
+    // select aggregation
+    await testSubjects.click('aggregationNameA');
+    await testSubjects.click('aggregationTypeSelect');
+    // assert all options are available
+    await find.byCssSelector('option[value="avg"]');
+    await find.byCssSelector('option[value="min"]');
+    await find.byCssSelector('option[value="max"]');
+    await find.byCssSelector('option[value="sum"]');
+    await find.byCssSelector('option[value="count"]');
+    await find.byCssSelector('option[value="cardinality"]');
+    await find.byCssSelector('option[value="p99"]');
+    await find.byCssSelector('option[value="p95"]');
+    await find.byCssSelector('option[value="rate"]');
+
+    // set first aggregation
+    await find.clickByCssSelector(`option[value="avg"]`);
+    const input1 = await find.byCssSelector('[data-test-subj="aggregationField"] input');
+    await input1.type('metricset.rtt');
+    await testSubjects.click('o11yClosablePopoverTitleButton');
+
+    // set second aggregation
+    await testSubjects.click('thresholdRuleCustomEquationEditorAddAggregationFieldButton');
+    await testSubjects.click('aggregationNameB');
+    await testSubjects.setValue('ruleKqlFilterSearchField', 'service.name : "opbeans-node"');
+    await testSubjects.click('o11yClosablePopoverTitleButton');
+
+    // set custom equation
```

Comment:
I couldn't figure out how to test the custom equation field, at least not quickly figure it out.

I encountered a element not intractable error. If anyone is a wizard in debugging those, let me know.